### ISSUE TITLE
Missing observation for @JmsListener response messages

### ISSUE THREAD

**Issue by jghandour** [spring-projects/spring-framework#33221]
When using @JmsListener that is configured with an ObservationRegistry can see that the ObservationRegistry is used for the incoming message but not utilized for the outgoing response message.  Specifically, the 'traceparent' header is not included in the response.   This was using Spring JMS 6.1.10 via Spring Boot 3.3.1.

A simple workout in the meantime is to explicitly use a JmsTemplate configured with ObservationRegistry to send a response.

```
@Bean
DefaultJmsListenerContainerFactory jmsContainerFactory(ConnectionFactory connectionFactory, ObservationRegistry registry) {
DefaultJmsListenerContainerFactory factory = new DefaultJmsListenerContainerFactory();
factory.setConnectionFactory(connectionFactory)
factory.setObservationRegistry(registry);
return factory;
}

@JmsListener("someQueue")
Message<String> handleMessage(Message message) { 
// Do stuff
return MessageBuilder.withPayload("test").build();
}
```

**Comment by bclozel** [comment]
Thanks for your report @jghandour - this should be available in a few minutes with 6.1.12-SNAPSHOTs.

**Comment by jghandour** [comment]
Excellent, thank you for the quick turn around!
